### ISSUE TITLE
Возможность импортировать не в глобальную область видимости

### ISSUE THREAD

**Issue by nixel2007** [EvilBeaver/OneScript#253]
Библиотеки постепенно наращивают функциональность, надеюсь, и количество их будет увеличиваться :) Да и возможность подключать файлы os напрямую тоже способствует.

Рано или поздно это может привести к конфликту имен процедур/классов/модулей.

Предлагаю обсудить возможность импортировать библиотеки в заданный namespace.

Что-то вроде:
`#Использовать v8runner Как МагияСКонфигуратором`

Таким образом создание экземпляра класса УправлениеКонфигуратором будет выглядеть вот так:
`Конфигуратор = Новый МагияСКонфигуратором.УправлениеКонфигуратором`

Обращаю внимание, что я говорю именно про импорт библиотек. Одиночные скрипты вполне успешно загружаются методами ПодключитьСценарий и ЗагрузитьСценарий

**Comment by nixel2007** [comment]
И да, я понимаю, что это вероятно большая переделка движка. Но хотя бы подумать и обсудить это стоит уже сейчас

**Comment by EvilBeaver** [comment]
В современных типовых сотни тысяч строк и неймспейсы до сих пор не понадобились. Переделка потенциально не очень большая, однако не думаю, что сие на данный момент востребовано.

**Comment by dmpas** [comment]
@EvilBeaver В современных типовых нет пользовательских классов. Тогда бы точно начался раздрай в именах посреди бесчисленных нетленок :smile:

**Comment by EvilBeaver** [comment]
да я понимаю.. )

**Comment by nixel2007** [comment]
@EvilBeaver 

> В современных типовых сотни тысяч строк и неймспейсы до сих пор не понадобились. Переделка потенциально не очень большая, однако не думаю, что сие на данный момент востребовано.

мне кажется, ты немного про другое. В современных типовых в принципе глобальный контекст не захламлен, т.к. глобальные модули можно пересчитать по пальцам одной руки, включая модули приложения.
Все остальные процедуры и функции сосредоточены в рамках общих модулей или контекстов объектов/менеджеров/форм.

#Использовать в oscript же добавляет процедуры и функции в глобальный контекст. Мы совсем недавно нарвались на это с режимом выгрузки файлов/конфигурации в gitsync и в v8runner. По сути то, что в gitsync назвали перечисление по другому - счастливая случайнось

**Comment by EvilBeaver** [comment]
да я же уже согласился вроде как. Просто не срочно

**Comment by asosnoviy** [comment]
Вот и нарвались. Класс `РезультатЗапроса`. Есть в библиотеке sql и в веб.движке

**Comment by EvilBeaver** [comment]
да. Причем есть и решение. Допилить sql в виде NuGet пакета и пакетом включать в веб-движок. Получится единая кодовая база. Сейчас в веб-движке так и так не подключается sql из-за каких то проблем с совместимостью формата DLL
